envi/memory.py

```diff
@@ -538,8 +546,18 @@ def writeMemory(self, va, bytes):
                 mva, msize, mperms, mfname = mmap
                 if not (mperms & MM_WRITE or self._supervisor):
                     raise envi.SegmentationViolation(va)
+
                 offset = va - mva
-                mapdef[3] = mbytes[:offset] + bytes + mbytes[offset+len(bytes):]
+                byteslen = len(bytes)
+                maxwritelen = msize - offset
+                if byteslen > maxwritelen:
+                    # if we're writing past the end of this map, recurse to find the next map
+                    # perms checks for that map will be performed, and size, etc... and if
+                    # an exception must be thrown, future writeMemory() can throw it
+                    mapdef[3] = mbytes[:offset] + bytes[:maxwritelen]
+                    self.writeMemory(mva + msize, bytes[maxwritelen:])
```

Comment:
with this change, a portion of the split write may succeed while a later portion fails and raises the exception. the side effects of the first write are not rejected. i think this behavior makes sense; because of its nuance, i'd recommend that it be documented, as well.

Reply:
i can extend the comment.

Reply:
did one better:  updated the docstr as well as the exception message.